src/main/java/org/commonjava/indy/service/tracking/data/cassandra/CassandraTrackingQuery.java

```diff
@@ -130,6 +131,37 @@ public void init()
         logger.info( "-- Cassandra Folo Records Keyspace and Tables created" );
     }
 
+    private ResultSet executeSession ( BoundStatement bind )
+    {
+        boolean exception = false;
+        ResultSet trackingRecord = null;
+        try
+        {
+            if ( session == null )
+            {
+                client.close();
+                client.init();
+                this.init();
+            }
+            trackingRecord = session.execute( bind );
```

Comment:
Not sure if the session need to be a new one or not, for example in tracking service `cluster.connect()` may try to reconnect.  But you can test locally if you have the instance for test before.

Reply:
Or I see, there is `client.init()`. so let's have a try.

Reply:
@sswguo Yeah, the session refetch will call through this class init method. I've tested locally, it could reconnect immediately after re-startup if relocated db server to a different IP, but still need to validate in the real devel Cluster env, I'll merge this to have a try.